### PR TITLE
Issue/#19 integrate logging framework

### LINKED ISSUE #19: Integrate Chosen Logging Framework
Now that we've chosen a logging framework, it's time to integrate it into our project. This task involves setting up the logging framework in our project, configuring it, and updating parts of our application to use it.

Tasks include:

- **Integration**: Integrate the chosen logging framework into our project. This involves adding necessary dependencies and setting up a basic configuration.
- **Usage**: Update parts of our application to use the new logging framework. This can be as simple as replacing System.out.println() calls with logging calls.
- **Testing**: Ensure that logging is working as expected in various parts of our application.

**Acceptance Criteria**:

1. The chosen logging framework has been integrated into our project.
2. Parts of our application have been updated to use the new logging framework.
3. Logging is working as expected in our application.
4. Dependency Injection used

### PR DESCRIPTION
**Pull request by FreakyF**
Closes #19

### CODE REVIEW

File: src/main/java/org/weathertrack/Main.java [hunk 1]
```diff
@@ -1,7 +1,19 @@
...
 	public static void main(String[] args) {
-		System.out.println("Hello and welcome!");
+		String logInfo = "Example log from {}";
+		for(int i = 0; i < 10000; i++) {
+			logger.trace(logInfo, Main.class.getSimpleName());
+			logger.info(logInfo, Main.class.getSimpleName());
+			logger.debug(logInfo, Main.class.getSimpleName());
+			logger.warn(logInfo, Main.class.getSimpleName());
+			logger.error(logInfo, Main.class.getSimpleName());
+		}
```
Comment (Tymisko) [inline src/main/java/org/weathertrack/Main.java:17]: Please remove this code.

File: src/main/resources/logback.xml
Comment (Tymisko) [inline src/main/resources/logback.xml:1]: Please remove redundant comments

Comment (Tymisko) [inline src/main/resources/logback.xml:1]: Please add a filter for the console and file to log at different levels. 
Files won't need as many logs as the console need.

**Review by Tymisko — CHANGES_REQUESTED** [review]

File: src/main/resources/logback.xml
Comment (Tymisko) [inline src/main/resources/logback.xml:1]: Set minimum logging level for the file to **WARN** and for the console to **INFO**.

Comment (Tymisko) [reply src/main/resources/logback.xml:1]: Apologize for the previous misunderstanding.

**Review by Tymisko — CHANGES_REQUESTED** [review]

**Review by Tymisko — CHANGES_REQUESTED** [review]
Please create generic `Logger<T>` interface with following methods:
```java
    void info(String message);
    void warn(String message);
    void error(String message);
    void info(String message, Throwable t);
    void warn(String message, Throwable t);
    void error(String message, Throwable t);
    void info(String format, Object... args);
    void warn(String format, Object... args);
    void error(String format, Object... args);
``` 
create also abstract class `AbstractLogger<T>` which will implement these interface methods as abstract and will have a protected final field with the name of the type specified.
It should also have a protected method `protected String format message (String message)` that will join the class name and message to log. 

And create an implementation for your specific chosen logging framework.
Its implementation name should explicitly describe which framework you are using.

It also should be derived class of `AbstractLogger<T>`.

**Comment by FreakyF** [comment]
> Please create generic `Logger<T>` interface with following methods:
> 
> ```java
>     void info(String message);
>     void warn(String message);
>     void error(String message);
>     void info(String message, Throwable t);
>     void warn(String message, Throwable t);
>     void error(String message, Throwable t);
>     void info(String format, Object... args);
>     void warn(String format, Object... args);
>     void error(String format, Object... args);
> ```
> 
> create also abstract class `AbstractLogger<T>` which will implement these interface methods as abstract and will have a protected final field with the name of the type specified. It should also have a protected method `protected String format message (String message)` that will join the class name and message to log.
> 
> And create an implementation for your specific chosen logging framework. Its implementation name should explicitly describe which framework you are using.
> 
> It also should be derived class of `AbstractLogger<T>`.


How about Trace and Debug levels?

File: src/main/java/org/weathertrack/util/logger/LogbackLogger.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package org.weathertrack.util.logger;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogbackLogger<T> extends BaseLogger<T> {
+	private final Logger logger;
+
+	public LogbackLogger(Class<T> className) {
```
Comment (Tymisko) [inline src/main/java/org/weathertrack/util/logger/LogbackLogger.java:9]: `className` is not a valid paramter name. `Class<T>` is not a string.

**Review by Tymisko — CHANGES_REQUESTED** [review]

**Review by Tymisko — APPROVED** [review]

File: .idea/sonarlint/issuestore/0/b/0b94eed6ac949458337e2826f9ae5d20679c8636
Comment (Tymisko) [inline .idea/sonarlint/issuestore/0/b/0b94eed6ac949458337e2826f9ae5d20679c8636:1]: Please add `**/.idea/sonarlint/` to `.gitignore`

**Review by Tymisko — CHANGES_REQUESTED** [review]

**Review by Tymisko — APPROVED** [review]

**Pull request merged**